ui/lib/kv/addon/components/kv-list-filter.js

```diff
@@ -13,42 +13,59 @@ import { tracked } from '@glimmer/tracking';
 
 /**
  * @module KvListFilter
- * `KvListFilter` component is used to filter through the KV metadata LIST response. It allows users to search through the current list, navigate into directories, and use keyboard functions to: autocomplete, view a secret, create a new secret, or clear the input field.
- *
- * @param {object} model - The adapter model object which contains an array of secret models.
+ * `KvListFilter` filters through the KV metadata LIST response. It allows users to search through the current list, navigate into directories, and use keyboard functions to: autocomplete, view a secret, create a new secret, or clear the input field.
+ *  *
+ * <KvListFilter
+ *  @secrets={{this.model.secrets}}
+ *  @mountPoint={{this.model.mountPoint}}
+ *  @filterValue="beep/my-"
+ *  @pageFilter="my-"
+ * />
+ * @param {array} secrets - An array of secret models.
  * @param {string} mountPoint - Where in the router files we're located. For this component it will always be vault.cluster.secrets.backend.kv
  * @param {string} filterValue - A concatenation between the list-directory's dynamic path "path-to-secret" and the queryParam "pageFilter". For example, if we're inside the beep/ directory searching for any secret that starts with "my-" this value will equal "beep/my-".
- * @param {string} pageFilter - The queryParam value.
+ * @param {string} pageFilter - The queryParam value, does not include pathToSecret ex: my-.
  */
 
 export default class KvListFilterComponent extends Component {
   @service router;
   @tracked filterIsFocused = false;
 
-  kvRoute(route) {
-    return `${this.args.mountPoint}.${route}`;
+  navigate(pathToSecret, pageFilter) {
+    const route = pathToSecret ? `${this.args.mountPoint}.list-directory` : `${this.args.mountPoint}.list`;
```

Comment:
curious why `mountPoint` has to be dynamic, elsewhere we just use `secret.backends...`

Reply:
It doesn't. In other engines, we've used it, but in KV we've just been passing in the hardcoded copy. LIST is the only area that uses it. It's a couple of changes to remove. So I'll make a clean up ticket and keep separate from this PR.

Reply:
For context. LIST was the first view I made and I was following a convention in kubernetes, but as we moved along it doesn't really make sense to have it as a param.